CorsixTH/Lua/dialogs/build_room.lua

```diff
@@ -145,6 +150,7 @@ function UIBuildRoom:setCategory(index)
     self.ui:tutorialStep(3, 3, 2)
   end
   self.category_index = index
+  self:updateList()
```

Comment:
I don't think you need this call here, as research_department calls across and you'll have the updated list of rooms already or you keep this call here and not call from research_department. Doing it from research you only do it once per discovered/available room and this does the update every category change.

Reply:
It works when it's just this line and not when it's just the line in research_department.